src/samplers/gibbs.jl

```diff
@@ -62,21 +77,19 @@ function Base.run(model, data, spl::Sampler{Gibbs})
         end
       elseif isa(local_spl, Sampler{PG})
         # Update new VarInfo to the reference particle
+        varInfo.index = 0
+        varInfo.num_produce = 0
         if ref_particle != nothing
-          ref_particle.vi.idcs = varInfo.idcs
-          ref_particle.vi.vals = varInfo.vals
-          ref_particle.vi.syms = varInfo.syms
-          ref_particle.vi.dists = varInfo.dists
-          ref_particle.vi.index = 0
-          ref_particle.vi.num_produce = 0
+          ref_particle.vi = varInfo
         end
-        # local samples
+        # Clean variables belonging to the current sampler
+        varInfo = retain(deepcopy(varInfo), local_spl.alg.group_id, 0)
```

Comment:
Doesn't this line empty the reference particle's `vals` field? If so, we might end up starting a new particle instead of replaying the reference particle.